brain/converter/crypto.py

```diff
@@ -0,0 +1,44 @@
+'''@crypto
+
+This file contains various cryptography wrappers.
+
+
+'''
+
+
+import os
+import hashlib
+import base64
+
```

Comment:
You can probably adjust the `saltlength` variable by implementing yaml definitions.  In this file, you could probably do:

```python
from flask import current_app

salt_length = current_app.config.get('SALT_LENGTH')
```

**Note:** kind of nitpicky, do you think we could change the `salt_length` variable to be changed from camel casing, to underscore casing?

which means you'll need to define the corresponding definition in the [`factory.py`](https://github.com/jeff1evesque/machine-learning/blob/12089fb8a6ec440cb4b257e3306400293e919592/factory.py#L60-L77), which then would need to be defined in the following yaml files:

- [`settings.yaml`](https://github.com/jeff1evesque/machine-learning/blob/12089fb8a6ec440cb4b257e3306400293e919592/hiera/settings.yaml)
- [`test/hiera/settings.yaml`](https://github.com/jeff1evesque/machine-learning/blob/12089fb8a6ec440cb4b257e3306400293e919592/hiera/test/hiera/settings.yaml)

Maybe create a new parent section called `authentication`, or add the definition in the existing `database` section.  I'm not 100% sure, since I'm not sure if the salt could be used in other aspects of the application.  Do you have ideas of where it may be used in other areas?  If so, maybe it'd be appropriate to create it's own section.

**Note:** the latter `settings.yaml` is called upon with docker builds (travis ci uses this for automated builds), while the former is used for vagrant builds (what you're using for development).

P.S. sorry for such a long reply for a short file :( Also, feel free to open corresponding issues related to this one, like database structure, and other things related to authentication you can think of.  In the meantime, I will either try to speed up solving the current [milestone 0.4](https://github.com/jeff1evesque/machine-learning/milestone/9), or determining ways to merge this pull request into another intermediate branch (not `master`).  The reason for this, is because I will create an official [0.4 release](https://github.com/jeff1evesque/machine-learning/releases), when it is completed, and that will be predicated on the git `master` branch.